sdks/java/core/src/main/java/org/apache/beam/sdk/values/RowWithGetters.java

```diff
@@ -123,4 +124,27 @@ public int getFieldCount() {
   public Object getGetterTarget() {
     return getterTarget;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null) {
+      return false;
+    }
+    if (o instanceof RowWithGetters) {
+      RowWithGetters other = (RowWithGetters) o;
+      return Objects.equals(getSchema(), other.getSchema())
+          && Objects.equals(getterTarget, other.getterTarget);
```

Comment:
Will this always produce results we want? I'm thinking aloud here...

Presumably our objective here is to determine if two `RowWithGetters` are equal. 

Since `getterTarget` is outside our control there are cases that we should consider:
 1. it might not have `equals` implemented correctly
 2. truly differing objects could produce rows that are equal for the scope of the schema (e.g. have extra fields ignored in the schema) 
 3. if the target is an array should it be using `Objects.deepEquals()`?

Reply:
1. The documentation for the currently supported getterTargets (JavaFieldSchema, JavaBeanSchema) currently declare that equals() is required to be declared. Longer term we can detect whether equals is declared, and if not we can generate a (slower) equals method.

2. This is an interesting point. For now I will document the requirements, when we handle this more generally per 1. I think we need to provide a way to deal with this.

3. Currently the target cannot be an array. It can only be a POJO or a Java Bean.

Reply:
Thanks - very helpful and agree on approach to document it. 
Suggest somewhere doc the user might need to add a distinct() for cases where they map partial rows in the source that can emit duplicate records.

Reply:
Actually an improper equals function can cause all sorts of havoc - not simply duplicate records (e.g. Rows might be stored in a container that relies on equality comparison). I think for now we should simply document that this case is not supported, and figure out how to support this case better (probably by writing code that recursively unrolls the Row).

Reply:
But these are very insightful points. Thanks for making them.

Reply:
Makes sense. Thanks.

Reply:
boosted documentation for now

Reply:
Where is that please?

Reply:
in JavaFieldSchema.java and JavaBeanSchema.java (as right now these are the classes that Beam users use when annotating Java classes for inference)

Reply:
Also added to SchemaRegistry.registerPOJO/registerJavaBean